ir/type.h

```diff
@@ -123,9 +128,38 @@ class VectorType final : public Type {
 };
 
 
+// Currently only supports aggregate type with two children
+// of int type.
+// Eg: {ix, i1}. Enforced in getTypeConstraints()
+class AggregateType final : public Type {
+  std::vector<Type*> childrenType;
+
+public:
+  AggregateType(std::string &&name) : Type(std::move(name)) {}
+  AggregateType(std::string &&name, std::initializer_list<Type*> types)
+    : Type(std::move(name)), childrenType(types) {
+    ENSURE(!childrenType.empty());
+  }
+
+  unsigned bits() const override;
+  smt::expr enforceAggregateType() const override;
+  smt::expr getTypeConstraints() const override;
+  smt::expr operator==(const AggregateType &rhs) const;
+  void fixup(const smt::Model &m) override;
+  smt::expr enforceIntOrPtrOrVectorType() const override;
+  void print(std::ostream &os) const override;
+
+  unsigned getChildrenSize() const { return childrenType.size(); }
```

Comment:
name is strange. Also there's casting going on. You can use auto as return type instead.
What about numElements()?